### PR TITLE
feat: `EcDoubleAssignNewYV1` hint

### LINKED ISSUE #386: EcDoubleAssignNewYV1
EcDoubleAssignNewYV1 hint  computes a new y-coordinate for doubling a point on an elliptic curve

Reference: 
https://github.com/starkware-libs/cairo-lang/blob/efa9648f57568aad8f8a13fbf027d2de7c63c2c0/src/starkware/cairo/common/cairo_secp/ec.cairo#L151

### PR DESCRIPTION
**Pull request by TAdev0**
Resolves #386 

@cicr99  as we discussed, i made a refactoring of `FastEcAddAssignNewY` and created a helper function `ComputeYCoordinate` so that both `EcDoubleAssignNewYV1` and `FastEcAddAssignNewY` use the same method.

### CODE REVIEW

File: pkg/hintrunner/zero/zerohint_utils.go [hunk 1]
```diff
@@ -45,3 +47,16 @@ func GetUint256AsFelts(vm *VM.VirtualMachine, ref hinter.ResOperander) (*fp.Elem
 
 	return low, high, nil
 }
+
```
Comment (MaksymMalicki) [inline pkg/hintrunner/zero/zerohint_utils.go:50]: could you place a short comment explaining what it does / provide the formula here as well?

Comment (TAdev0) [reply pkg/hintrunner/zero/zerohint_utils.go:50]: sure! thanks

**Comment by TAdev0** [comment]
@MaksymMalicki  done!

File: pkg/hintrunner/zero/zerohint_ec.go [hunk 2]
```diff
@@ -335,6 +283,51 @@ func createFastEcAddAssignNewXHinter(resolver hintReferenceResolver) (hinter.Hin
 	return newFastEcAddAssignNewXHint(slope, point0, point1), nil
 }
 
+// FastEcAddAssignNewY hint computes a new y-coordinate for fast elliptic curve addition
+//
+// `newFastEcAddAssignNewYHint` doesn't take any operander as argument
+// This hint follows the execution of `FastEcAddAssignNewX` hint when computing the addition of two given points
+// This is why all variables are already accessible in the current scope
```
Comment (rodrigo-pino) [inline pkg/hintrunner/zero/zerohint_ec.go:290]: Why are this description and ...

File: pkg/hintrunner/zero/zerohint_ec.go [hunk 3]
```diff
@@ -538,6 +531,52 @@ func createEcDoubleAssignNewXV1Hinter(resolver hintReferenceResolver) (hinter.Hi
 	return newEcDoubleAssignNewXV1Hint(slope, point), nil
 }
 
+// EcDoubleAssignNewYV1 hint computes a new y-coordinate when doubling a point
+// on an elliptic curve
+//
+// `newEcDoubleAssignNewYV1Hint` doesn't take any operander as argument
+// This hint follows the execution of `EcDoubleAssignNewXV1` hint when doubling a point
+// This is why all variables are already accessible in the current scope
```
Comment (rodrigo-pino) [inline pkg/hintrunner/zero/zerohint_ec.go:539]: ... this description different if the code performs the same operations

File: pkg/hintrunner/zero/zerohint_ec.go [hunk 4]
```diff
@@ -538,6 +531,52 @@ func createEcDoubleAssignNewXV1Hinter(resolver hintReferenceResolver) (hinter.Hi
...
+//
+// `newEcDoubleAssignNewYV1Hint` assigns the new y-coordinate as `value` in the current scope
+func newEcDoubleAssignNewYV1Hint() hinter.Hinter {
```
Comment (rodrigo-pino) [inline pkg/hintrunner/zero/zerohint_ec.go:542]: This two functions look awfully similar (I only spotted a `x0`, `y0` vs a `x`,`y` in the scope vars). Is there any way to make the two hints the same?

File: pkg/hintrunner/zero/hintcode.go [hunk 5]
```diff
@@ -64,11 +64,12 @@ const (
...
+	fastEcAddAssignNewYCode string = "value = new_y = (slope * (x0 - new_x) - y0) % SECP_P"
 	ecDoubleSlopeV1Code     string = "from starkware.cairo.common.cairo_secp.secp_utils import SECP_P, pack\nfrom starkware.python.math_utils import ec_double_slope\n\n# Compute the slope.\nx = pack(ids.point.x, PRIME)\ny = pack(ids.point.y, PRIME)\nvalue = slope = ec_double_slope(point=(x, y), alpha=0, p=SECP_P)"
 	computeSlopeV1Code      string = "from starkware.cairo.common.cairo_secp.secp_utils import SECP_P, pack\nfrom starkware.python.math_utils import line_slope\n\n# Compute the slope.\nx0 = pack(ids.point0.x, PRIME)\ny0 = pack(ids.point0.y, PRIME)\nx1 = pack(ids.point1.x, PRIME)\ny1 = pack(ids.point1.y, PRIME)\nvalue = slope = line_slope(point1=(x0, y0), point2=(x1, y1), p=SECP_P)"
 	ecDoubleAssignNewXV1    string = "from starkware.cairo.common.cairo_secp.secp_utils import SECP_P, pack\n\nslope = pack(ids.slope, PRIME)\nx = pack(ids.point.x, PRIME)\ny = pack(ids.point.y, PRIME)\n\nvalue = new_x = (pow(slope, 2, SECP_P) - 2 * x) % SECP_P"
+	ecDoubleAssignNewYV1    string = "value = new_y = (slope * (x - new_x) - y) % SECP_P"
```
[the remark this answers is not in the document]
Comment (TAdev0) [reply pkg/hintrunner/zero/hintcode.go:72]: No they are not

File: pkg/hintrunner/zero/zerohint_ec.go (hunk 3 above)
Comment (TAdev0) [reply pkg/hintrunner/zero/zerohint_ec.go:539]: What do you mean ? They borrow the same logique but one does addition of points , the other double the same point ?

File: pkg/hintrunner/zero/zerohint_ec.go (hunk 4 above)
Comment (TAdev0) [reply pkg/hintrunner/zero/zerohint_ec.go:542]: No we can't 
This is why I did this refactoring ! To only implement the logic once

**Comment by TAdev0** [comment]
@rodrigo-pino  needs approval to merge this one
we already discussed it on friday, its good to go

**Review by rodrigo-pino — APPROVED** [review]

**Pull request merged**